### PR TITLE
Deliver root mail

### LINKED ISSUE #947: Local System User Mail Delivery Issues
My setup is as follows, mail.maildomain.com, this domain does not have MX or any deliverable mailboxes, it is just a mail server for other virtual domains. This servers local mail (root@) appears to have delivery issues its trying to send mail to a hostname and domain name that do not accept mail, in addition to being the external hostname and domain name of the mail server itself. I added a .forward file in the /root/ directory. The address is an external email address that exists and is deliverable from the internet. Will this solve my issues? The host machine does not have postfix or any mail packages installed, it is all running in this docker container. What are opinions at a possible solution if what I did is in-sufficent?

Any feedback would be greatly appreciated. 

/etc/hosts
internetaddressofmailserver mail.mailserverdomain.com
127.0.0.1 mail.mailserverdomain.com localhost.localdomain localhost
127.0.0.1 mail

Log: 
mail postfix/smtpd[29932]: connect from unknown[172.20.0.1]
Apr 11 04:00:03 mail docker-compose[1438]: mail    | Apr 11 04:00:02 mail postfix/smtp[29921]: warning: host mail.mymailserverdomain.com[internetaddressofmailserver]:25 greeted me with my own hostname mail.mymailserverdomain.com
Apr 11 04:00:03 mail docker-compose[1438]: mail    | Apr 11 04:00:02 mail postfix/smtp[29921]: warning: host mail.mymailserverdomain.com[internetaddressofmailserver]:25 replied to HELO/EHLO with my own hostname mail.mymailserverdomain.com
Apr 11 04:00:03 mail docker-compose[1438]: mail    | Apr 11 04:00:02 mail postfix/smtp[29921]: B36AA3E80D: to=<root@mail.mymailserverdomain.com>, orig_to=<root>, relay=mail.mymailserverdomain.com[internetaddressofmailserver]:25, delay=0.65, delays=0.05/0.02/0.58/0, dsn=5.4.6, status=bounced (mail for mail.mymailserverdomain.com loops back to myself)
Apr 11 04:00:03 mail docker-compose[1438]: mail    | Apr 11 04:00:02 mail postfix/smtpd[29932]: disconnect from unknown[172.20.0.1] ehlo=1 quit=1 commands=2 00:00

### PR DESCRIPTION
**Pull request by crazystick**
Deliver root mail to POSTMASTER_ADDRESS, fixes #947 
If POSTMASTER_ADDRESS isn't defined, it defaults to postmaster@mydomain.com, and so you can also get it delivered by creating a virtual alias for postmaster@mydomain.com or via my favoured approach of using a regexp alias for postmaster@* (as seen in the test config).

One other thing - the tests were already failing for me before my change, as the email counts had increased by one, hence why I am increasing them by two in this PR.

### CODE REVIEW

File: test/config/postfix-regexp.cf [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /^test[0-9][0-9]*@localhost.localdomain/ user1@localhost.localdomain
 /^bounce.*@.*/ external1@otherdomain.tld
+/^postmaster@/ user1@localhost.localdomain
```
Comment (johansmitsnl) [inline test/config/postfix-regexp.cf]: Can you add a new line.

Comment (crazystick) [reply test/config/postfix-regexp.cf]: as I mentioned in the PR body, the tests are failing before my changes as the counts are already wrong - I assume some earlier change broke these tests.

**Comment by 17Halbe** [comment]
The last test (PR #942) succeeded with the current count. Any idea where the second mail is coming from? (Bounce, fail, something like that?)

**Comment by crazystick** [comment]
I can't see any extra emails. When I comment out my extra mail in Makefile#252 then the build fails for me. Perhaps it's a test that was sending mail to postmaster@somewhere that was previously not delivered to any mailbox...

**Comment by crazystick** [comment]
OK - I think figured it out. The extra email is the virus notification email which is sent to postmaster and previously was not delivered. Now it is thanks to the extra config in `postfix-regexp.cf`

**Comment by 17Halbe** [comment]
So this was a bug? The mail was supposed to get delivered to postmaster?

**Comment by crazystick** [comment]
Well more like an undocumented / untested feature! If you configured your postmaster address to get delivered, then you would get the email. It's unrelated to this PR, which does fix a bug (the bug is that mail to root is not delivered, and no configuration could fix that).

**Pull request merged**